lib/puppet/parser/functions/foreman.rb

```diff
@@ -64,14 +66,29 @@ module Puppet::Parser::Functions
 
     begin
       path = URI.escape("/api/#{item}?search=#{search}&per_page=#{per_page}")
-      uri = URI.parse(foreman_url)
 
       req = Net::HTTP::Get.new(path)
-      req.basic_auth(foreman_user, foreman_pass)
       req['Content-Type'] = 'application/json'
       req['Accept'] = 'application/json'
 
-      http = Net::HTTP.new(uri.host, uri.port)
+      if use_tfmproxy
+        configfile = if use_tfmproxy.is_a? String and File.exist?(use_tfmproxy)
```

Comment:
I'd move the ```File.exist?``` to just before reading it. If a user specifies a non-existing file it shouldn't silently ignore it. The same is true for ```true``` but ```/etc/foreman-proxy/settings.yml``` doesn't exist.

Reply:
you mean exception or just puts ?

Reply:
I think a hard exception is in order.